python/cudf/cudf/core/dataframe.py

```diff
@@ -1475,7 +1475,20 @@ def reindex(
 
         if idx is not None:
             idx = idx if isinstance(idx, Index) else as_index(idx)
-            if df.index.dtype != idx.dtype:
+
+            from cudf.core import MultiIndex
+
+            if type(idx) == MultiIndex:
+                idx_match = [
+                    x.any()
+                    for x in [
+                        df.index._source_data.dtypes == idx._source_data.dtypes
+                    ]
+                ][0]
+            else:
+                idx_match = [df.index.dtype == idx.dtype]
```

Comment:
Can you explain what `idx_match` is used for / controls?

Reply:
`reindex` did not work with a `MultiIndex`. So if it's an MI it compares the `dtypes` of the `df.index` and `idx`. Since there are multiple results from the comparison I get x.any() so I get a single boolean value to proceed as it would with a regular Index. It took a bit of spelunking to figure this out. If there is an easier way to get the `dtypes` for comparison in an MI I am interested.

The regular index behavior was `if df.index.dtype != idx.dtype` so I just tried to find the most minimal solution for that comparison with an MI.

Reply:
This appears to have produced some larger issues that I need to look at further. Investigating.